### ISSUE TITLE
MinGW has unistd.h but no pipe()

### ISSUE THREAD

**Issue by akheron** [akheron/jansson#426]
This causes the json_dumpfd() tests to fail to compile on MinGW.

**Comment by pond29** [comment]
Hi

**Comment by coreyfarrell** [comment]
Does MinGW use autotools?  I can add detection for ./configure to avoid this in the test but I can't help with CMake.

**Comment by akheron** [comment]
I think both autotools and CMake work in MinGW. Or maybe you need MSYS? I'm not that familiar with Windows.

**Comment by coreyfarrell** [comment]
I have zero familiarity with Windows.  How did you find this issue?  If someone has a build log I can investigate further.

**Comment by othotuld** [comment]
I have the same issue and end up here. I think if we don't want to skip the test, we can use _pipe function by utilizing this macro.

```
#ifdef __MINGW32__
#include <fcntl.h>
#define pipe(fds) _pipe(fds,1024, _O_BINARY)
#endif
```
 
1024 is buffer size, reserves for the pipe. it is the way, MS implements pipe function.

more info here.
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/pipe?view=vs-2019

**Comment by akheron** [comment]
Fixed in #500
